### PR TITLE
Add CLI task to get/set settings, refs #13604

### PR DESCRIPTION
**Pull request by mcantelon**
Added CLI task to get or set settings, optionally to or from files.

### CODE REVIEW

File: lib/task/tools/settingsTask.class.php [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+     * @param mixed $arguments
+     * @param mixed $options
+     */
+    public function execute($arguments = [], $options = [])
+    {
+        parent::execute($arguments, $options);
+
+        $this->validateOptions($arguments, $options);
+
+        $this->displayCurrentSettings();
```
Comment (djjuhasz) [inline lib/task/tools/settingsTask.class.php]: When I run the task I always get the full list of settings, e.g.

```
[dev/issue-13604-settings-task][~/atom]$ symfony tools:setting get version
------------------------------------------------------------------------------------
Name                                                              Scope
------------------------------------------------------------------------------------
_csrf_token                                                       element_visibility
accession_counter
accession_mask
accession_mask_enabled
actor                                                             default_template
...
```

Is this intentional? It makes it hard to actually see the value for the selected setting.

**Comment by jraddaoui** [comment]
Nice one @mcantelon!

@mamedin recently asked for something similar to get setting and config values, I created this quick "tools:run" task ...

```php
<?php

$config = sfConfig::getAll();
$config['factories'] = sfFactoryConfigHandler::getConfiguration(
    $this->configuration->getConfigPaths('config/factories.yml')
);

echo json_encode($config, JSON_PRETTY_PRINT) . "\n";
```

I think it could be a good addition to include a way to get the values from the factories config, maybe not in this task or without a set option. And also to add an option to output as JSON when printing multiple values, so you can then pipe the output a bit easily ...

`php symfony tools:run output_config_json.php | jq -r '.factories.user.param.timeout'`

File: lib/task/tools/settingsTask.class.php (hunk 1 above)
Comment (djjuhasz) [reply lib/task/tools/settingsTask.class.php]: The "_csrf_token" setting is due to a bug (https://projects.artefactual.com/issues/13613) I just discovered while testing this task. 🤦

Comment (jraddaoui) [reply lib/task/tools/settingsTask.class.php]: Oh, good catch @djjuhasz!

Comment (mcantelon) [reply lib/task/tools/settingsTask.class.php]: Weird one, that "_csrf_token". 😨 

Yeah, I added that display of a list of settings to make the settings easier to work with, but displaying it the way I'm displaying it - with every command - is kludge-y. I'll take it out for now... I could make a new CLI task to display settings if need be. That'd be less weird.

**Comment by mcantelon** [comment]
Thanks @radda! Good thinking... I've kind of shoehorned settings display into this task so I think I'm going to remove it and add the logic to a new task. Adding display of factory settings makes sense, too, so I'll add a Redmine issue to describe this new task (and specify that it should have JSON output too).

File: lib/task/tools/settingsTask.class.php [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+                'en'
+            ),
+            new sfCommandOption(
+                'scope',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'Scope of setting to get or set',
+                null
+            ),
+            new sfCommandOption(
```
Comment (djjuhasz) [inline lib/task/tools/settingsTask.class.php:324]: Opinion: I think it's more intuitive to make the 'value' an optional argument, so the command can be called like `symfony tools:setting set [name] [value]`

File: lib/task/tools/settingsTask.class.php [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+                'strict',
+                null,
+                sfCommandOption::PARAMETER_NONE,
+                'Prevent creation of new settings when performing "set" operation',
+                null
+            ),
+        ]);
+
+        $this->namespace = 'tools';
+        $this->name = 'setting';
```
Comment (djjuhasz) [inline lib/task/tools/settingsTask.class.php]: Nitpick: I think it's better for the class name to be `SettingTask` or to make `$this->name = 'settings';` for consistency.

**Review by djjuhasz — APPROVED** [review]
@mcantelon a few comments but no blockers.  I'm impressed that you simulated the database interactions for the tests. :)

File: lib/task/tools/settingsTask.class.php (hunk 3 above)
Comment (mcantelon) [reply lib/task/tools/settingsTask.class.php]: Good call. Changing the name to "settings" makes sense.

**Comment by mcantelon** [comment]
@djjuhasz I've updated the PR and did a bit of minor refactoring, fixed some errors I found, added a couple of new tests, and got rid of the `to-file` and `from-file` options (replacing them with a single `file` option).

(Oh you approved it earlier... thanks! I'll merge it!)

**Pull request merged**